datasets/chooser.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from django.utils.translation import gettext_lazy as _
+from wagtail import hooks
+
+from generic_chooser.views import ModelChooserMixin, ModelChooserViewSet
+from generic_chooser.widgets import AdminChooser
+
+from kausal_common.datasets.models import Dataset
+
+from users.models import User
+
+if TYPE_CHECKING:
+    from django.http import HttpRequest
+
+
+class DatasetChooserMixin(ModelChooserMixin):
+    order_by: str | list[str]
+    request: HttpRequest
+
+    def get_unfiltered_object_list(self):
+        user = self.request.user
+        if not isinstance(user, User):  # could be anonymous or some unexpected implementation of AbstractBaseUser
+            return Dataset.objects.none()
+        objects = Dataset.permission_policy().instances_user_has_permission_for(user, 'view')
+        if self.order_by:
+            if isinstance(self.order_by, str):
+                objects = objects.order_by(self.order_by)
+            else:
+                objects = objects.order_by(*self.order_by)
```

Comment:
@bbliem I am not sure if it's worth it but this is currently copy-pasted from the superclass, and I think we could achive the same by calling super and then using one of the methods in the permission policy which create a Q object which could be used to filter that queryset from the superclass.